service/src/main/java/bio/terra/cbas/runsets/results/RunCompletionHandler.java

```diff
@@ -112,21 +110,18 @@ public RunCompletionResult updateResults(
         if (recordAttributes != null && !recordAttributes.isEmpty()) {
           String errorMessage = saveOutputsToWDS(updatableRun, recordAttributes);
           if (errorMessage != null && !errorMessage.isEmpty()) {
-            errors.add(errorMessage);
+            // Failure to update WDS. Let this be retried by Cromwell
+            logger.error(errorMessage);
+            return RunCompletionResult.ERROR;
```

Comment:
TOL: Should we set an attempt limit on this retry? So after N retries for this error CBAS eventually records an error? Or is that not worth it since in second/third attempt it output saving to WDS should succeed?

Reply:
Cromwell will give up after 10 attempts. After that the user will trigger a smartpoller retry every refresh. 

So, as long as WDS is unavailable, the method will stay in Running but if WDS ever does come back up the workflow will move to Done the next time the user refreshes the page.

Of course, WDS _should_ come back within a couple of retries so I wouldn't expect us to ever actually reach that final state, but there is at least some resolution there.

Reply:
Good to know. Thank you for the clarification!